src/globus_cli/parsing/param_types/omittable.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import annotations
+
+import datetime
+import typing as t
+import uuid
+
+import click.types
+import globus_sdk
+
+from globus_cli._click_compat import (
+    OLDER_CLICK_API,
+    shim_get_metavar,
+    shim_get_missing_message,
+)
+
+
+class OmittableInt(click.ParamType):
```

Comment:
None of these custom param types are documented & they're pretty high level concepts that developers will be asked to wield properly.

I think each of them need a docstring explaining behaviorally what to expect if used + probably a module-level docstring that explains what this category of classes is.